be/src/connector/hive_connector.cpp

```diff
@@ -261,7 +264,13 @@ void HiveDataSource::_init_tuples_and_slots(RuntimeState* state) {
             _equality_delete_slots.emplace_back(d_slot_desc);
             if (!id_to_slots.contains(d_slot_desc->id())) {
                 _materialize_slots.push_back(d_slot_desc);
-                _materialize_index_in_chunk.push_back(delete_column_index++);
+                const auto& it = id_to_index.find(d_slot_desc->id());
+                if (it == id_to_index.end()) {
+                    return Status::InternalError(
+                            "Invalid slot id in delete_column_slot_ids. id =  " + std::to_string(d_slot_desc->id()) +
```

Comment:
Could you use fmt::format instead?

Reply:
And I want to suggest that change `_materialize_slots.push_back(d_slot_desc);` after this check succeeds.